db/indexes_test.go

```diff
@@ -57,53 +59,63 @@ func TestInitializeIndexes(t *testing.T) {
 
 			// Make sure we can drop and reinitialize twice
 			for i := 0; i < 2; i++ {
-				dropErr := base.DropAllIndexes(base.TestCtx(t), n1qlStore)
-				require.NoError(t, dropErr, "Error dropping all indexes")
-
-				initErr := InitializeIndexes(n1qlStore, test.xattrs, 0, true)
-				require.NoError(t, initErr, "Error initializing all indexes")
-
-				// Recreate the primary index required by the test bucket pooling framework
-				err := n1qlStore.CreatePrimaryIndex(base.PrimaryIndexName, nil)
+				err := dropAndInitializeIndexes(base.TestCtx(t), n1qlStore, b, test.xattrs)
+				require.NoError(t, err, "Error dropping and initialising all indexes on bucket")
+			}
+			// check to see if current indexes match what is expected by the rest of the test
+			// if not we drop and reinitialize these indexes using the overall test environment variables for XATTRS
+			err := validateAllIndexesOnline(b, base.TestUseXattrs())
+			if err != nil {
+				err = dropAndInitializeIndexes(base.TestCtx(t), n1qlStore, b, base.TestUseXattrs())
 				require.NoError(t, err)
-
-				validateErr := validateAllIndexesOnline(b)
-				require.NoError(t, validateErr, "Error validating indexes online")
 			}
 		})
 	}
 
 }
 
 // Reset bucket state
-func validateAllIndexesOnline(bucket base.Bucket) error {
-
-	n1QLStore, ok := base.AsN1QLStore(bucket)
-	if !ok {
-		return fmt.Errorf("Bucket is not gocb bucket: %T", bucket)
-	}
-
-	// Retrieve all indexes
-	getIndexesStatement := fmt.Sprintf("SELECT indexes.name, indexes.state from system:indexes where keyspace_id = %q", n1QLStore.GetName())
-	results, err := n1QLStore.Query(getIndexesStatement, nil, base.RequestPlus, true)
+func validateAllIndexesOnline(bucket base.Bucket, xattrs bool) error {
+	col, err := base.AsCollection(bucket)
 	if err != nil {
 		return err
 	}
 
-	var indexRow struct {
-		Name  string
-		State string
+	cluster := col.GetCluster()
+	mgr := cluster.QueryIndexes()
+
+	watchOption := gocb.WatchQueryIndexOptions{
+		WatchPrimary:   true,
+		ScopeName:      col.ScopeName(),
+		CollectionName: col.Name(),
```

Comment:
This happily passes with CBS 6.6, which surprised me since it doesn't use collections. If you are adding code that adds gocb parameters for collections, it's worth trying to remember to pass through jenkins with couchbase server 6.6. We do run this weekly.